tests/translator/output/error_embedded_connectors.json

```diff
@@ -1,3 +1,3 @@
nd: 23. Transforming resource with id [DuplicateTest] attempts to create a new resource with id [DuplicateTestDuplicateId] and type \"AWS::Serverless::Connector\". A resource with that id already exists within this template. Please use a different id for that resource. Resource with id [EventsRuleMissingLambdaFunctionArn] is invalid. Unable to get Lambda function ARN from 'Destination' resource. Resource with id [EventsRuleMissingSnsTopicArn] is invalid. Destination.Arn is missing. Resource with id [EventsRuleMissingSqsQueueUrl] is invalid. Unable to create connector from AWS::Events::Rule to AWS::SQS:Queue; it's not supported or the template is invalid. Resource with id [MyFunctionBothIdAndOtherProps] is invalid. Must provide 'Id' (with optional 'Qualifier') or a supported combination of other properties. Resource with id [MyFunctionEmptyDestination] is invalid. Missing required property 'Destination'. Resource with id [MyFunctionEmptyPermissions] is invalid. Missing required property 'Permissions'. Resource with id [MyFunctionEmptyPermissionsList] is invalid. 'Permissions' cannot be empty; valid values are: Read, Write. Resource with id [MyFunctionEmptyProperties] is invalid. Attribute 'MyFunctionEmptyProperties.Properties' should be a map. Resource with id [MyFunctionMissingRoleMissingRole] is invalid. Unable to get IAM role name from 'Source' resource. Resource with id [MyFunctionNoDestination] is invalid. Missing required property 'Destination'. Resource with id [MyFunctionNoDictConnectorNonDictConnector] is invalid. MyFunctionNoDictConnector.MyFunctionNoDictConnectorNonDictConnector should be a map. Resource with id [MyFunctionNoIdMissingType] is invalid. 'Type' is missing or not a string. Resource with id [MyFunctionNoPermissions] is invalid. Missing required property 'Permissions'. Resource with id [MyFunctionNoProperties] is invalid. Attribute 'MyFunctionNoProperties.Properties' should be a map. Resource with id [MyFunctionNoStrId] is invalid. 'Id' is missing or not a string. Resource with id [MyFunctionNonDictProperties] is invalid. Attribute 'MyFunctionNonDictProperties.Properties' should be a map. Resource with id [MyFunctionNonExistentId] is invalid. Unable to find resource with logical ID 'ThisDoesntExist'. Resource with id [MyFunctionTestSourceReferenceId] is invalid. 'Id' shouldn't be defined in 'SourceReference'. Resource with id [MyFunctionUnsupportedAccessCategory] is invalid. Unsupported 'Permissions' provided for connector from AWS::Lambda::Function to AWS::SQS::Queue; valid values are: Read, Write. Resource with id [MyQueueMissingRoleDestination] is invalid. Unable to get IAM role name from 'Destination' resource. Resource with id [MyQueueUnsupportedAccessCategoryCombination] is invalid. Unsupported 'Permissions' provided for connector from AWS::SQS::Queue to AWS::Lambda::Function; valid combinations are: Read + Write. Resource with id [MyResourceWithoutTypeResourceWithoutType] is invalid. 'Type' is missing or not a string."
```

Comment:
this becomes really difficult to spot the diff 👁️

Reply:
Yea not sure why the diff looks like that. Fwiw the error message is `Resource with id [MyFunctionNoProperties] is invalid. Attribute 'MyFunctionNoProperties.Properties' should be a map.`

Reply:
> Attribute 'MyFunctionNoProperties.Properties'

this feels off. It should be 
> Resource with id `MyFunction` is invalid. `Connectors.Properties` should be a map.

Reply:
hmm but `MyFunctionNoProperties` is the "name" of the connector (`MyFunction` being the source logical id and `NoProperties` being the connector name). If we have that, wouldn't it be easier for them find the connector thats causing the error?

Reply:
I sort of agree. It helps identify where the issue is faster.

Reply:
> MyFunctionNoProperties

If customers do not understand how connector work, can they understand what `MyFunctionNoProperties` is?

Reply:
`MyFunctionNoProperties` is the logical id they input themself right?

Reply:
NoProperites is the logical id they input themselves. MyFunctionNoProperites helps them identify the source resource (`MyFunction`) and the embedded connector with the issue (`NoProperties`). Maybe `MyFunction.NoProperties` would be better?

Reply:
Still off. the error message says:

> Resource with id [MyFunctionNoProperties] is invalid.

but customers cannot find such resource in their template

Reply:
> MyFunctionNoProperties is the logical id they input themself right?

@GavinZZ no check the test yaml

Reply:
Can we make this "Resource with id MyFunction is invalid. Connectors.Properties should be a map." as it is super clear and actionable

Reply:
`Connector.Properties` is too general imo :/ if they have a lot of connectors attached to one source resource then it would be difficult to find the error one. `Resource with id MyFunction is invalid. NoProperties.Properties should be a map.` might be the best middle ground?

Reply:
yeah please